internals/overlord/servstate/handlers.go

```diff
@@ -588,6 +588,9 @@ func (s *serviceData) exited(exitCode int) error {
 			s.transition(stateStopped)
 		}
 
+	case stateStopped:
```

Comment:
This is a drive-by fix. It doesn't actually change any behaviour, but avoids an unnecessary info log in this situation (presumably primarily during test runs).